j2v/generation/generator.py

```diff
@@ -153,5 +155,20 @@ def __add_dimension(self, field_path_sql, current_view, dimension_name, dim_val,
         self.views_dimensions_expr[current_view].add(new_dimension)
 
         sql_select = st.field_str_template.format(__path=field_path_sql, TABLE=current_view,
-                                                  json_type=json_type, path_alias=full_path_nice.upper())
+                                              json_type=json_type, path_alias=full_path_nice.upper())
         self.all_fields[current_view].add(sql_select)
+
+        if json_type.lower() == "number":
+            sql_no_nulls_select = st.non_nullable_numeric_field_str_template.format(__path=field_path_sql,
```

Comment:
Maybe we can simplify this logic by having just 1 string template and a map with a mapping of (json_type, dim_type) to a default value for NULLs? Then we can expand it easily in the future for other types.